### DISCUSSION TITLE
相談

### DISCUSSION THREAD

**Discussion by t-sakurai816** [t-sakurai816/inYourFreeTime#11] [Ideas]
相談事一覧

**Comment by t-sakurai816** [comment]
# ブランチについて

ブランチについてgit-flowやgithub-flowを参考にしました。今回は簡略化のために、以下でどうでしょう？
@yCroma 

* main
  * 本番リリース後まで使わない。リリース後はこのブランチがデプロイされているものとする
* develop
  * 開発時のブランチ。基本的にはこれが最新版になる。
* dev/#{issue num}
  * `dev/#1`のようにissue番号をもとにissueを潰していくブランチ

**Reply by yCroma** [reply]
devの方針には賛成、ただ#3のような感じでブランチを切れたか覚えてないから確認したい。

**Reply by t-sakurai816** [reply]
`dev/#3`のようにはブランチ切れます！俺がいつもそうしてた
`#3`オンリーはダメかもしれないけど

**Reply by yCroma** [reply]
開発ブランチの名前を`/dev`から`/feature`にしたいです。理由としては、ブランチを移動する際に、`/develop`と`/dev`があって補完が中途半端になるからです。

featureは機能って意味の英単語。機能毎にブランチを切って実装するから、その単語が使われてる気がする。

**Reply by t-sakurai816** [reply]
OKです！
`feature/#4`のようにしましょう。

**Comment by t-sakurai816** [comment]
# 名称について

本番、開発と環境を分けると思いますが、プログラムで使うときの名前を統一しましょう

| 環境 | 名前 | 略称 |
| --- |  --- | --- |
| 本番 | production | prod | 
| 開発 | staging | stg |

**Reply by t-sakurai816** [reply]
やはりこっちにしません？
prdでいい気がしてきた

| 環境 | 名前 | 略称 |
| --- |  --- | --- |
| 本番 | production | prd | 
| 開発 | staging | stg |

**Comment by t-sakurai816** [comment]
# ホスティング先について

ホスティング先について、以下のように考えていますが、大丈夫でしょうか？
GitHubPagesを開発段階で利用する理由は、複雑な設定をせずともデプロイができるからです。
S3にデプロイするにはmainブランチの`frontend`ディレクトリ内を手動デプロイとします。（ゆくゆくはGitHubActionsにしましょう

本番：S3
開発：GitHubPages

**Reply by yCroma** [reply]
これに関しては、可能ならNetlifyが安全。
理由はGitHubPagesは、リポジトリから作成した時、ドキュメントルートが/リポジトリ名になって、/ではなくなるから。
SPAを構築した際、絶対パスで制御される可能性があって、その際にドキュメントルートを再設定しないといけない可能性があるから。

ダメではないけど、不安材料が消せたら理想みたいな感じ

**Reply by yCroma** [reply]
S3ホスティングするなら、パスがズレる可能性があるので、予めGitHubPagesで練習します。
やったことはあるので、上手くいくかだけって感じ。多分出来ると思う。

**Reply by t-sakurai816** [reply]
あーーーーそういうことか！理解できました。
Netlifyにしましょうか

S3ホスティングは t-sakurai816/s3-hosting があるので、これを応用できればなーという感じです。

**Reply by t-sakurai816** [reply]
S3ホスティングには欠点がありまして、
index.htmlを返すように設定していたとします。

hoge.com/　：　OK
hoge.com/dir/　：　NG
hoge.com/dir/index.html　：　OK

のようになります。最初のドメインのみのときは`index.html`いらないけど、以降ディレクトリが変わるときは`index.html`が必要になります

**Reply by yCroma** [reply]
SPAなんで、極論index.htmlがあればOK
ただ、SPAって、リクエストされたパスに応じて表示する中身を切り替えるんだけど、その時のパスがホスティングによってズレるかどうかで変わる感じ

**Reply by t-sakurai816** [reply]
ほー…なるほど
問題がでるようだったら、他の方法も考えよう。

**Reply by yCroma** [reply]
質問でーす。
HTMLとして静的ホスティングしますか？それともNodeで配信して良いんですか？どちらでもオッケーです！

**Reply by yCroma** [reply]
静的ホスティングできるようにビルドしまーす。お騒がせしましたー

**Comment by yCroma** [comment]
# フロントエンドのマージについて

黒馬の方針としては、一度確認してもらったほうが良いと思っている。ただ、このまま続けていくと形式的に確認するという事態にもなる気がする。黒馬個人で進めていくのか、適度なところで折り合いつけるのか、そこら辺をどうしようかと気になっている。

**Reply by t-sakurai816** [reply]
とりあえず完成を優先させるために、フロントについてはレビューなしにしましょうー
ゴリゴリ作っちゃってください

**Reply by yCroma** [reply]
今見ましたーりょうかーい
